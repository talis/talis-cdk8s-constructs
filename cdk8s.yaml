language: typescript
imports:
  - k8s@1.30.0
