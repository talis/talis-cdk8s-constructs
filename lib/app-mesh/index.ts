export * from "./ingress-gateway-nlb";
export * from "./app-service";
