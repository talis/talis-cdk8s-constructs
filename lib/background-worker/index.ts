export * from "./background-worker-props";
export * from "./background-worker";
