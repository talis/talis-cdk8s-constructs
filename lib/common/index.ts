export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
