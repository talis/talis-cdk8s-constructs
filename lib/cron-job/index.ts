export * from "./cron-job-props";
export * from "./cron-job";
