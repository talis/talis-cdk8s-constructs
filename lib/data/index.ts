export * from "./config-map";
export * from "./secret";
export * from "./factory";
