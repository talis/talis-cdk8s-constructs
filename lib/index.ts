export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
