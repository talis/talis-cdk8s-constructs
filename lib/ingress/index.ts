export * from "./ingress-props";
export * from "./ingress";
