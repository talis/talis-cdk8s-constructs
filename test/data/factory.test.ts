import { Testing } from "cdk8s";
import { createImagePullSecret } from "../../lib";

describe("factory", () => {
  describe("createImagePullSecret", () => {
    test("Creates config with Docker Hub credentials by default", () => {
      const chart = Testing.chart();
      createImagePullSecret(chart, { auth: "user:test" });
      const results = Testing.synth(chart);
      expect(results).toMatchSnapshot();
    });

    test("Creates config with custom registry", () => {
      const chart = Testing.chart();
      createImagePullSecret(chart, {
        auth: "username:password",
        registry: "https://registry.example.io/",
      });
      const results = Testing.synth(chart);
      expect(results).toMatchSnapshot();
    });

    test("Creates config with auth as-is", () => {
      const chart = Testing.chart();
      createImagePullSecret(chart, {
        auth: "QVdTOnNvbWUtc2VjcmV0LUVDUi1rZXk=",
        encode: false,
        registry: "123456789000.dkr.ecr.eu-west-1.amazonaws.com",
      });
      const results = Testing.synth(chart);
      expect(results).toMatchSnapshot();
    });

    test("Lets specify node id", () => {
      const chart = Testing.chart();
      const secret = createImagePullSecret(chart, {
        id: "my-secret",
        auth: "user:test",
      });
      expect(secret.node.id).toBe("my-secret");
    });
  });
});
