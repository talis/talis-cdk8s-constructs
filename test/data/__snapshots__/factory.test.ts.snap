// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`factory > createDockerHubSecretFromEnv > Creates config with Docker Hub credentials from env vars 1`] = `
[
  {
    "apiVersion": "v1",
    "data": {
      ".dockerconfigjson": "eyJhdXRocyI6eyJodHRwczovL2luZGV4LmRvY2tlci5pby92MS8iOnsiYXV0aCI6ImMyOXRaWFZ6WlhJNmMyVmpjbVYwTVRJeiJ9fX0=",
    },
    "kind": "Secret",
    "metadata": {
      "name": "test-docker-hub-cred-c8d251b2",
    },
    "type": "kubernetes.io/dockerconfigjson",
  },
]
`;

exports[`factory > createDockerHubSecretFromEnv > Throws if Docker credentials env vars are not set 1`] = `"DOCKER_USERNAME and DOCKER_PASSWORD must be set in the environment"`;

exports[`factory > createImagePullSecret > Creates config with Docker Hub credentials by default 1`] = `
[
  {
    "apiVersion": "v1",
    "data": {
      ".dockerconfigjson": "eyJhdXRocyI6eyJodHRwczovL2luZGV4LmRvY2tlci5pby92MS8iOnsiYXV0aCI6ImRYTmxjanAwWlhOMCJ9fX0=",
    },
    "kind": "Secret",
    "metadata": {
      "name": "test-image-pull-secret-c859bf04",
    },
    "type": "kubernetes.io/dockerconfigjson",
  },
]
`;

exports[`factory > createImagePullSecret > Creates config with auth as-is 1`] = `
[
  {
    "apiVersion": "v1",
    "data": {
      ".dockerconfigjson": "eyJhdXRocyI6eyIxMjM0NTY3ODkwMDAuZGtyLmVjci5ldS13ZXN0LTEuYW1hem9uYXdzLmNvbSI6eyJhdXRoIjoiUVZkVE9uTnZiV1V0YzJWamNtVjBMVVZEVWkxclpYaz0ifX19",
    },
    "kind": "Secret",
    "metadata": {
      "name": "test-image-pull-secret-c859bf04",
    },
    "type": "kubernetes.io/dockerconfigjson",
  },
]
`;

exports[`factory > createImagePullSecret > Creates config with custom registry 1`] = `
[
  {
    "apiVersion": "v1",
    "data": {
      ".dockerconfigjson": "eyJhdXRocyI6eyJodHRwczovL3JlZ2lzdHJ5LmV4YW1wbGUuaW8vIjp7ImF1dGgiOiJkWE5sY201aGJXVTZjR0Z6YzNkdmNtUT0ifX19",
    },
    "kind": "Secret",
    "metadata": {
      "name": "test-image-pull-secret-c859bf04",
    },
    "type": "kubernetes.io/dockerconfigjson",
  },
]
`;
